rules/tock.bzl

```diff
@@ -115,56 +121,35 @@ def _tock_image_impl(ctx):
     )
 
     ctx.actions.run(
-        outputs = [images[0]],
-        inputs = [kernel_binary],
+        outputs = [image],
+        inputs = [kernel_binary] + [app[TockApplication].tbf for app in ctx.attr.apps],
         arguments = [
-            "flash",
-            "--board",
-            "opentitan_earlgrey",
-            "--flash-file",
-            images[0].path,
-            "--address",
-            "0x20000000",
-            kernel_binary.path,
-        ],
-        executable = ctx.executable._tockloader,
+            "--rcfile=",
+            "image",
+            "assemble",
+            "--mirror=false",
+            "--output={}".format(image.path),
+            "--size=0x70000",
+            "{}@0".format(kernel_binary.path),
+        ] + ["{}@{}".format(app[TockApplication].tbf.path, app[TockApplication].flash_start - 0x20010000) for app in ctx.attr.apps],
```

Comment:
I think this constant should also be expressed in a parameter (although, in this case, its probably ok for the parameter to have this value as a default - it's unlikely to change unless you're working with a chip that has a different flash arrangement or you're intentionally creating a bad image for a test case).

Reply:
I made the address range of the Tock image into new `image_start` and `image_end` parameters, to avoid the magic numbers this section currently has.